modules/mono/glue/GodotSharp/GodotPlugins/PluginLoadContext.cs

```diff
@@ -21,6 +21,26 @@ public class PluginLoadContext : AssemblyLoadContext
             _resolver = new AssemblyDependencyResolver(pluginPath);
             _sharedAssemblies = sharedAssemblies;
             _mainLoadContext = mainLoadContext;
+
+            if (string.IsNullOrEmpty(AppContext.BaseDirectory))
+            {
+                // See https://github.com/dotnet/runtime/blob/v6.0.0/src/libraries/System.Private.CoreLib/src/System/AppContext.AnyOS.cs#L17-L35
+                // but Assembly.Location is unavailable, because we load assemblies from memory.
+                string? baseDirectory = Path.GetDirectoryName(pluginPath);
+                if (baseDirectory != null)
+                {
+                    if (!Path.EndsInDirectorySeparator(baseDirectory))
+                        baseDirectory += Path.PathSeparator;
+                    // This SetData call effectively sets AppContext.BaseDirectory
+                    // See https://github.com/dotnet/runtime/blob/v6.0.0/src/libraries/System.Private.CoreLib/src/System/AppContext.cs#L21-L25
+                    AppDomain.CurrentDomain.SetData("APP_CONTEXT_BASE_DIRECTORY", baseDirectory);
+                }
+                else
+                {
+                    // TODO: How to log from GodotPlugins? (delegate pointer?)
+                    Console.Error.WriteLine("Failed to set AppContext.BaseDirectory. Dynamic loading of libraries may fail.");
```

Comment:
Regarding the TODO, `Console.Error` is fine. It's what we currently use in `GodotPlugins/Main.cs` when an exception is caught before returning to native code.

Reply:
Yeah, thats where I copied it from, I included the TODO just so that this line doesn't get lost if someone ever implements butter logging.